core/lib/dal/src/models/storage_sync.rs

```diff
@@ -55,7 +55,7 @@ impl TryFrom<StorageSyncBlock> for SyncBlock {
                     .try_into()
                     .decode_column("l1_batch_number")?,
             ),
-            last_in_batch: block.last_batch_miniblock == Some(block.number),
+            last_in_batch: block.tx_count == 0,
```

Comment:
Fictive l2 block sealing and L1 batch sealing are not happening in the same transaction anymore, so the only way to determine if l2 block is the last in batch is to check if it has 0 txs.